### PR TITLE
login component created

### PR DESCRIPTION
**Pull request by lape15**
image

### CODE REVIEW

File: src/Pages/Auth/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect } from "react";
+import { useFormik } from "formik";
+import { Link } from "react-router-dom";
+import * as Yup from "yup";
+import { TimelineLite, Power3 } from "gsap";
+import { makeAuthenticatedApiCall } from "../Api.js";
+
+const url = "https://atumaatu.herokuapp.com/v1/sessions/";
```
Comment (Nkemjiks) [inline src/Pages/Auth/Login.jsx]: Can you move this URL to the api.js file. It is consistent and not going to change. The only thing you will need to pass with be the `sessions` part. You can rename to route

```suggestion
const path = "sessions";
```

File: src/App.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ function App(props) {
       ) : null}
       <Switch>
         <Route exact path="/" component={Landing} />
-        <Route exact path="/register" component={Register} />
+        <Route path="/(login|register)/" exact component={Auth} />
```
Comment (Nkemjiks) [inline src/App.js:24]: Great job figuring this out. I learnt something new today

File: src/Pages/Api.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import axios from "axios";
+
+const makeAuthenticatedApiCall = (method, url, data) => {
+  return axios({
+    method: method,
+    url: url,
```
Comment (Nkemjiks) [inline src/Pages/Api.js]:
```suggestion
    url: "https://atumaatu.herokuapp.com/v1/${path}/",
```

File: src/Pages/Auth/index.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        repeat: -1,
+        ease: "steps(14)",
+        borderRightColor: "rgba(255,255,255,0)",
+      },
+      0
+    );
+    tl.play();
+  }, []);
+
+  if (localStorage.user) {
```
Comment (Nkemjiks) [inline src/Pages/Auth/index.js]: I didn't know about this. i had to ask my boyfriend to explain. Wooow. You are doing great

File: src/Pages/Auth/Login.jsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        .required("Required"),
+
+      password: Yup.string()
+        .min(6, "Password is too short")
+        .required("Required"),
+    }),
+
+    onSubmit: (values) => {
+      setLoading(true);
+      const responses = makeAuthenticatedApiCall("post", url, {
```
Comment (Nkemjiks) [inline src/Pages/Auth/Login.jsx]: So this part will now be

```suggestion
      const responses = makeAuthenticatedApiCall("post", path, {
```

File: src/Pages/Auth/Login.jsx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        email: values.email,
+        password: values.password,
+      });
```
Comment (Nkemjiks) [inline src/Pages/Auth/Login.jsx:50]: Spaces....

File: src/Pages/Auth/Login.jsx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+const Login = (props) => {
+  const [loading, setLoading] = useState(false);
+  const [user, setUser] = useState(null);
+  const [authenticationError, setAuthenticationError] = useState(null);
+
+  if (user) {
+    props.history.push("/dashboard");
+    localStorage.setItem("user", JSON.stringify(user.token));
+  }
```
Comment (Nkemjiks) [inline src/Pages/Auth/Login.jsx:18]: Spaces..

**Review by Nkemjiks — CHANGES_REQUESTED** [review]
Great job on this one. Just small changes

File: src/Pages/Auth/Login.jsx [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+  let tl = new TimelineLite();
+
+  useEffect(() => {
+    tl.from(".login", 3, { y: 40, ease: Power3.easeOut });
+  }, []);
```
Comment (Nkemjiks) [inline src/Pages/Auth/Login.jsx]:
```suggestion
  }, [tl]);
```

File: src/Pages/Auth/index.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+      { borderRightColor: "rgba(255,255,255,0.8)" },
+      {
+        repeat: -1,
+        ease: "steps(14)",
+        borderRightColor: "rgba(255,255,255,0)",
+      },
+      0
+    );
+    tl.play();
+  }, []);
```
Comment (Nkemjiks) [inline src/Pages/Auth/index.js]:
```suggestion
  }, [tl]);
```

File: src/Pages/Auth/Register.jsx [hunk 10]
```diff
@@ -1,20 +1,27 @@
...
   const [loading, setLoading] = useState(false);
+
   if (user) {
     props.history.push("/dashboard");
     localStorage.setItem("user", JSON.stringify(user.token));
   }
+  let tl = new TimelineLite();
+  useEffect(() => {
+    tl.from(".register", 3, { y: 10, ease: Power3.easeOut });
+  }, []);
```
Comment (Nkemjiks) [inline src/Pages/Auth/Register.jsx]:
```suggestion
  }, [tl]);
```

File: src/Pages/Auth/Register.jsx [hunk 11]
```diff
@@ -1,20 +1,27 @@
...
 const Register = (props) => {
   const [authenticationError, setAuthenticationError] = useState(null);
   const [user, setUser] = useState(null);
   const [loading, setLoading] = useState(false);
+
   if (user) {
     props.history.push("/dashboard");
     localStorage.setItem("user", JSON.stringify(user.token));
   }
+  let tl = new TimelineLite();
```
Comment (Nkemjiks) [inline src/Pages/Auth/Register.jsx:21]:
```suggestion
  let tl = new TimelineLite();
  
```

File: src/Pages/Auth/Login.jsx [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+          <span className="error">
+            <i className="fas fa-times"></i>
+            {formik.errors.password}
+          </span>
+        ) : null}
+      </div>
+      <div className="w-80">
+        <button type="submit" disabled={loading}>
+          Login
+          {loading && <i class="fas fa-spinner"></i>}
```
Comment (Nkemjiks) [inline src/Pages/Auth/Login.jsx]:
```suggestion
          {loading && <i className="fas fa-spinner"></i>}
```

File: src/Pages/Auth/index.js [hunk 13]
```diff
@@ -0,0 +1,92 @@
+import React, { useEffect } from "react";
```
Comment (Nkemjiks) [inline src/Pages/Auth/index.js]: This file should be `index.jsx` not `index.js`

File: src/Pages/Auth/index.js [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+
+  if (localStorage.user) {
+    return <Redirect to="/dashboard" />;
+  }
+
+  return (
+    <div className="auth">
+      <div className="side">
+        <Link to="/" className="img-link">
+          <img src={logo} alt="BudgetIt logo" />{" "}
```
Comment (Nkemjiks) [inline src/Pages/Auth/index.js]:
```suggestion
          <img src={logo} alt="BudgetIt logo" />
```

File: src/Pages/Auth/Login.jsx (hunk 1 above)
Comment (lape15) [reply src/Pages/Auth/Login.jsx]: Ohk

File: src/App.js (hunk 2 above)
Comment (lape15) [reply src/App.js:24]: Thank you

**Pull request merged**